core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -92,4 +95,44 @@ public static Map<String, String> propertiesWithPrefix(
         .filter(e -> e.getKey().startsWith(prefix))
         .collect(Collectors.toMap(e -> e.getKey().replaceFirst(prefix, ""), Map.Entry::getValue));
   }
+
+  public static Map<String, String> applySchemaChanges(
+      Map<String, String> properties,
+      List<String> deletedColumns,
+      Map<String, String> renamedColumns,
+      Set<String> columnPrefixProperties) {
```

Comment:
Nit: what do you think to change to prefixProperties or columnProperties?  columnPrefixProperties may indicate wrongly that column is the prefix, so its a bit misleading at first.

Reply:
Changed to `columnProperties`.